conans/client/graph/graph_binaries.py

```diff
@@ -233,29 +196,50 @@ def _process_node(self, node, pref, build_mode, update, remotes):
-            package_layout = self._cache.pkg_layout(latest_prev_for_pkg_id)
+        if not cache_latest_prev:
+            try:
+                remote, remote_prev, prev_time = self._get_package_from_remotes(node, pref, remotes, update)
+            except NotFoundException:
+                node.binary = BINARY_MISSING
+                node.prev = None
+                node.binary_remote = None
+            else:
+                node.binary = BINARY_DOWNLOAD
+                node.prev = remote_prev.revision
+                node.binary_remote = remote
+        else:
+            package_layout = self._cache.pkg_layout(cache_latest_prev)
             self._evaluate_clean_pkg_folder_dirty(node, package_layout, pref)
+            if update:
+                try:
+                    remote, remote_prev, prev_time = self._get_package_from_remotes(node, pref,
+                                                                                    remotes, update)
+                except NotFoundException:
+                    output.warn("Can't update, no package in remote")
+                except NoRemoteAvailable:
+                    output.warn("Can't update, no remote defined")
+                else:
+                    cache_time = self._cache.get_timestamp(cache_latest_prev)
+                    # TODO: cache 2.0 should we update the date if the prev is the same?
+                    if cache_time < prev_time and cache_latest_prev != remote_prev:
+                        node.binary = BINARY_UPDATE
+                        node.prev = remote_prev.revision
+                        node.binary_remote = remote
+                        output.info("Current package revision is older than the remote one")
+                    else:
+                        node.binary = BINARY_CACHE
+                        node.binary_remote = None
+                        node.prev = cache_latest_prev.revision
+                        output.info("Current package revision is newer than the remote one")
 
-        remote = remotes.selected
-        remote_selected = remote is not None
-
-        if not remote:
-            # If the remote_name is not given, follow the binary remote, or the recipe remote
-            # If it is defined it won't iterate (might change in conan2.0)
-            remote_name = self._cache.get_remote(latest_prev_for_pkg_id or pref.ref)
-            remote = remotes.get(remote_name)
-
-        if latest_prev_for_pkg_id:  # Binary already exists in local, check if we want to update
-            node.prev = latest_prev_for_pkg_id.revision
-            self._evaluate_cache_pkg(node, latest_prev_for_pkg_id, remote, remotes, update)
-        else:  # Binary does NOT exist locally
-            # Returned remote might be different than the passed one if iterating remotes
-            remote = self._evaluate_remote_pkg(node, pref, remote, remotes, remote_selected)
-
-        node.binary_remote = remote
+        if not node.binary:
```

Comment:
This should belong to the ``else`` clause in line 213 (the ``if`` clearly always define ``node.binary``)